substrate/frame/revive/src/exec.rs

```diff
@@ -290,6 +290,14 @@ pub trait PrecompileExt: sealing::Sealed {
 			.adjust_gas(charged, RuntimeCosts::Precompile(actual_weight));
 	}
 
+	/// Charges the gas meter with the given token or halts execution if not enough gas is left.
+	fn charge_or_halt<Tok: crate::gas::Token<Self::T>>(
+		&mut self,
+		token: Tok,
+	) -> ControlFlow<crate::vm::evm::Halt, crate::gas::ChargedAmount> {
+		self.gas_meter_mut().charge_or_halt(token)
+	}
```

Comment:
Why is this necessary? Isn't charge already supposed to stop execution when not enough gas? Can't the `ControlFlow` be created by the caller?